src/zenml/zen_stores/schemas/artifact_schemas.py

```diff
@@ -147,11 +134,20 @@ def to_model(
             # This is an old source which was simply an importable source path
             data_type = Source.from_import_path(self.data_type)
 
+        producer_step_run_id = None
+
+        for step_run in self.output_of_step_runs:
+            if step_run.step_run.status == ExecutionStatus.COMPLETED:
+                producer_step_run_id = step_run.step_run.id
+                break
+
         return ArtifactResponseModel(
             id=self.id,
             name=self.name,
             artifact_store_id=self.artifact_store_id,
-            user=self.user.to_model() if self.user else None,
+            user=self.user.to_model(_block_recursion=True)
+            if self.user
+            else None,
```

Comment:
Same here, let's tackle hydration separately
```suggestion
            user=self.user.to_model() if self.user else None,
```